Add basic test bundles and documentation (#78)
hexatomic/hexatomic: docs/dev/src/development/automated-tests/ui-integration-tests.md

class TestCorpusStructure {
  
  private SWTWorkbenchBot bot = new SWTWorkbenchBot(ContextHelper.getEclipseContext());

  @Test
  void testRenameDocument() {
    // Make sure to activate the part to test before selecting SWT components
    bot.partById("org.corpus_tools.hexatomic.corpusedit.part.corpusstructure").show();
    
    // Add corpus graph 1 by clicking on the first toolbar button ("Add") in the corpus structure editor part
    bot.toolbarDropDownButton(0).click();
    // Add corpus 1
    bot.toolbarDropDownButton(0).click();
    // Add document_1
    bot.toolbarDropDownButton(0).click();
    // Select and edit the first document
    bot.tree().getTreeItem("corpus_graph_1").getNode("corpus_1").getNode("document_1").select();
    bot.tree().getTreeItem("corpus_graph_1").getNode("corpus_1").getNode("document_1").doubleClick();
    bot.text("document_1").setText("abc").pressShortcut(Keystrokes.LF);
   
    // Make sure that the document has been renamed
    bot.tree().expandNode("corpus_graph_1").expandNode("corpus_1").expandNode("abc");
    assertNotNull(bot.tree().getTreeItem("corpus_graph_1").getNode("corpus_1").getNode("abc"));
  }
}

```

> To test whether an action was successful, you can't access the internal data model of the application.

Reviewer: ```suggestion
To test whether an action was successful, integration tests do not access the internal data model of the application.
```

Wording.

> Thus, you have to check the effects on the user interface (like the result of renaming is a new tree item with the 

Reviewer: ```suggestion
Instead, they check the effects on the user interface (for example, the result of a renaming action is a new tree item with the 
```

Wording

new name).
To test internal states, use unit tests for the controllers of the bundle instead.

When your test case has more than one test, all tests are executed in the same environment.

> Thus, previous executed tests will influence the next ones.

Reviewer: ```suggestion
Thus, previously executed tests will influence subsequent ones.
```

Wording.

> You can set the execution order a test explicitly: <https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-execution-order>.

Reviewer: ```suggestion
You can set the execution order within a test class explicitly with the `@Order(int)` annotation: <https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-execution-order>.
```

Add more info about ordering, so that not everyone will have to check the docs.

> If you add an integration to a new bundle, that was not tested before, you have to add the bundle manually to the 

Reviewer: ```suggestion
If you integrate a new bundle that has not been tested before, you have to add the bundle manually to the 
```

Wording
